src/platform/common/variables/customEnvironmentVariablesProvider.node.ts

```diff
@@ -94,30 +116,17 @@ export class CustomEnvironmentVariablesProvider implements ICustomEnvironmentVar
             return cacheStoreIndexedByWorkspaceFolder.data!;
         }
 
-        const workspaceFolderUri = this.getWorkspaceFolderUri(resource);
-        if (!workspaceFolderUri) {
-            traceInfoIfCI(`No workspace folder found for ${resource ? resource.fsPath : '<No Resource>'}`);
-            return;
-        }
         this.trackedWorkspaceFolders.add(workspaceFolderUri.fsPath || '');
 
-        const envFile = this.getEnvFile(workspaceFolderUri, purpose);
+        const envFile = this.getEnvFile(workspaceFolderUri);
         this.createFileWatcher(envFile, workspaceFolderUri);
 
         const promise = this.envVarsService.parseFile(envFile, process.env);
         promise.then((result) => (cacheStoreIndexedByWorkspaceFolder.data = result)).ignoreErrors();
         return promise;
     }
-    public configurationChanged(e: ConfigurationChangeEvent) {
```

Comment:
Just revert a lot of the changes that were made in a previous PR (reverting a revert commit)